data/service/data-stream.js

```diff
@@ -357,6 +357,16 @@ exports.DataStream = DataProvider.specialize(/** @lends DataStream.prototype */
             stream.query = selector;
             return stream;
         }
+    },
```

Comment:
Why not have withTypeOrSelector states a deprecation message (only once) and then call the new withTypeOrQuery, no need to have twice the code

Reply:
Good point! Done. 

https://github.com/montagejs/montage/pull/1992/commits/b23207fdb2a83b2b41e169dbd4f54f098250a4ae